src/java.base/share/classes/java/util/Arrays.java

```diff
@@ -5444,7 +5444,9 @@ public static void parallelSetAll(double[] array, IntToDoubleFunction generator)
      * @since 1.8
      */
     public static <T> Stream<T> stream(T[] array, int startInclusive, int endExclusive) {
-        return StreamSupport.stream(spliterator(array, startInclusive, endExclusive), false);
+        var spliterator = spliterator(array, startInclusive, endExclusive);
+        if (startInclusive == endExclusive) return Stream.empty();
```

Comment:
Can't we just add the `if` statement to before the `spliterator` is computed?

Reply:
Thanks @liach for the suggestion. The spliterator serves the purpose of checking the arguments. For example, if array is null, the method should throw a NPE. Similarly, startInclusive and endExclusive have to be in the range of the array length. If we swap around the lines, someone could call stream(null, -100, -100) and it would happily return an empty stream.

Furthermore, the empty streams can inherit characteristics of a spliterator. In the code you pointed out, I don't do that, but will change that. It makes it much easier to keep the empty streams consistent with their previous behavior. One might argue that it is pointless to call distinct() sorted() and unordered() on an empty stream and expect it to change, but it was easy enough to implement so that we can keep consistency FWIW.